src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx

```diff
@@ -223,7 +241,7 @@ export default function ConsumerIncentivesHomeScreen() {
 
   const onPressCTA = async () => {
     if (canClaimRewards) {
-      dispatch(claimRewards(superchargeRewards))
+      dispatch(claimRewards())
```

Comment:
Why not continuing to pass the rewards?
I find it better to avoid global state when possible.

Reply:
what prompted this is in the slice, the definition of `claimRewards` has an unused variable :/ i then saw that we are passing the rewards as an action payload only to use it in the saga. can you clarify why it's better to avoid global state? the saga already reads from the store, are we saying that it's better to reduce the number of selectors used?

it also seemed more error prone to respond to an action payload, when that payload comes from the store anyway so there might be mismatches depending on when the action was dispatched, e.g. we fetch the supercharge rewards on the wallet home screen, when users come into the supercharge screen another fetch is kicked off. if the responses are different for whatever reason, and the user presses claim with the payload of the first response before the second response comes back, then they'd claim rewards that are different to what we expect? idk how real this scenario is, but it felt like

Reply:
This is more in general. I know the supercharge code is handled quite globally for now in Redux.

The difference is that we're acting exactly on what the user did in the component: "claim this specific reward".
Which is a good thing. As it can't be impacted by another requests changing the value under its feet.

> supercharge rewards on the wallet home screen, when users come into the supercharge screen another fetch is kicked off. if the responses are different for whatever reason, and the user presses claim with the payload of the first response before the second response comes back, then they'd claim rewards that are different to what we expect? idk how real this scenario is, but it felt like

Yes and I'd expect the initial reward to be claimed.
But like you said, it's not a real scenario.

However if we take the example of WC requests or sending transactions. It helps to see possible problems from using values from the global state that can change anytime.

Scenario:
- App receives WC request to approve, stores it in single global variable
- User sees the screen to approve
- User clicks to approve
- Before the approve can be processed, app receives another requests and changes global variable
- Result is the wrong requests gets approved, which could have really bad consequences.

Reply:
thanks for the explanation! makes sense. like you said, i'm not sure that this scenario is likely to impact supercharge because of the global implementation but it sounds like this change isn't necessarily an improvement so i reverted it. it wasn't necessary anyway